src/coord/src/coord.rs

```diff
@@ -738,9 +749,50 @@ where
                 // We use an `if let` here because the peek could have been cancelled already.
                 // We can also potentially receive multiple `Complete` responses, followed by
                 // a `Dropped` response.
-                if let Some(channel) = self.pending_tails.get_mut(&sink_id) {
+                if let Some(PendingTail {
+                    channel,
+                    emit_progress,
+                    object_columns,
+                }) = self.pending_tails.get_mut(&sink_id)
+                {
```

Comment:
Maybe makes no sense, but could this be bundled into a method on `PendingTail`, to better encapsulate the complexity?

Reply:
Done.